common.h

```diff
@@ -0,0 +1,20 @@
+#ifndef __COMMON_H__
+#define __COMMON_H__
+
+
+/* Useful macros*/
+
+#define STR_NOEXPAND(x) #x     // This does not expand x.
+#define STR(x) STR_NOEXPAND(x) // This will expand x.
+
+
+/* Target platform properties. */
+
+#define BITS_IN_BYTE 8
+#define WORD_TYPE uint32_t
```

Comment:
Please have a look at http://stackoverflow.com/questions/10265842/signed-and-unsigned-machine-word-types

IMHO It'd be sufficient to declare `word_t` as `intptr_t`. This would ensure that on common 32 and 64 bit architectures `word_t` has size of machine word, which is the most efficient type to represent bitset.